### ISSUE TITLE
Speech and mic on/off control

### ISSUE THREAD

**Issue by puripuneet** [Microsoft/BotFramework-WebChat#859]
It seems that by default speech turns on only when the microphone icon is manually clicked -- in other words, the bot speaks to the user only after the user clicks on the microphone icon to speak to it first.  This doesn't work well in some scenarios, for example, when we want the bot to speak out a greeting at startup, instead of just displaying it as text.

Also, the inputHint value of "expectingInput" from the bot framework only turns on the microphone for a couple of seconds.  How to make the mic stay on longer, or ideally until inputHint value of "acceptingInput" or "ignoringInput" is received from the bot framework?

**Comment by DanielTizon** [comment]
Same problem here! 

I would like that bot start speaking by voice without need that user use "mic" button before.

Is it possible to do it?

**Comment by compulim** [comment]
@puripuneet sounds great idea.


Currently, we don't have API to control the microphone button programmatically (unless you hack into our Redux store and send a `Listening_Starting` action). Always-on microphone sounds great idea.

For the second issue, when we detect silence for [5 seconds](https://github.com/Microsoft/BotFramework-WebChat/blob/master/src/Store.ts#L656), we will stop listening, regardless whether it is related to `expectingInput` or not. Maybe we can add an option there.

**Comment by puripuneet** [comment]
@compulim the answer is "yes" to your question "For the first problem, do you mean TTS (text-to-speech) is only available when the microphone is clicked for this first time?"

**Comment by matmatos** [comment]
@compulim  Any updates on the always-on option for the microphone?

**Comment by GayathriR2018** [comment]
Is there any update on this. We need to enable the mic on chat bot for intervals of time also. Is there any workarounds for this?

**Comment by compulim** [comment]
BTW, we coded a "speech button" sample that show how you can write new UI and use API to turn on/off microphone. It's powered by Web Chat minus the UI.

You can play the live demo at https://microsoft.github.io/BotFramework-WebChat/breakaway-speech-button/, and source code at https://github.com/Microsoft/BotFramework-WebChat/tree/preview/samples/breakaway-speech-button.

**Comment by SanaWaseem** [comment]
The links are broken

**Comment by corinagum** [comment]
Links fixed.

**Comment by adalfa** [comment]
please update us on the issue raised  by @puripuneet to TTS part, currently we have no OOTB method to turn it on by default in for example, a welcome card.

**Comment by AndreMantas** [comment]
Demo is broken, check console after clicking on mic:

`WebSocket connection to 'wss://westus.stt.speech.microsoft.com/speech/recognition/interactive/cognitiveservices/v1?language=pt-PT&format=detailed&Authorization=eyJhbGciOiJodHRwOi8vd3d3LnczLm9yZy8yMDAxLzA0L3htbGRzaWctbW9yZSNobWFjLXNoYTI1NiIsInR5cCI6IkpXVCJ9(...)' failed: HTTP Authentication failed; no valid credentials available`

Also, current implementation fails to speak text activities in at least 2 cases:

a) when bot sends proactive activities (like welcome message)

b) when user uploads a file and bot replies

**How can this be solved?
Is there some store event that we can dispatch?**

---

Also, side question if anyone can help me... I need to integrate webchat with google speech API (their recognition and synthesis is better at least for my language...), where can I look for pointers on how to implement a ponyfill for easily integrate with renderWebChat and pass the webSpeechPonyfillFactory?

**Comment by corinagum** [comment]
@AndreMantas regarding the error you are receiving on sample 13, could you file a new bug? Please follow the template and that will be a great help. 

For speech not speaking text activities, I would suggest filing another issue for that as well, maybe as a Question. We will have to investigate what's going on.

In regards to your implementation questions, we recommend asking these on [Stack Overflow](https://stackoverflow.com/tags/botframework)). We have engineers that monitor our botframework tag and can help you out.

Thank you!

**Comment by collegewap** [comment]
Updated links:

https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/06.recomposing-ui/b.speech-ui

https://microsoft.github.io/BotFramework-WebChat/06.recomposing-ui/b.speech-ui/
